File: ui/lib/client/startedConsumer.dart
class StartedConsumer {
  StartedConsumer(
      this.id,
      this.config,
      this.startedAtEpoch
      );

  String id;
  String config;
  int startedAtEpoch;

  Map<String, Object> get asJson {
    return {
      'id': id,
      'config': config,
      'startedAtEpoch': startedAtEpoch
    };
  }

  static StartedConsumer fromJson(Map<String, dynamic> json) {
    return StartedConsumer(
        json['id'],
        json['config'],
        json['startedAtEpoch']);
  }
}

File: client/jvm/src/main/scala/expressions/client/kafka/StartedConsumer.scala
package expressions.client.kafka

import io.circe.Codec

/**
  * Keep track of our running consumers
  * @param id
  * @param config
  * @param startedAtEpoch
  */
case class StartedConsumer(id: String, config: String, startedAtEpoch: Long)

object StartedConsumer {
  given codec : Codec[StartedConsumer] = io.circe.generic.semiauto.deriveCodec[StartedConsumer]
}
